src/app/content/components/Page/contentLinkHandler.ts

```diff
@@ -37,39 +43,65 @@ const isPathRefernceForBook = (pathname: string, book: Book) => (ref: PageRefere
       || ('uuid' in ref.params.book && ref.params.book.uuid === book.id)
     );
 
-export const contentLinkHandler = (anchor: HTMLAnchorElement, getProps: () => ContentLinkProp) => (e: MouseEvent) => {
-  const {references, navigate, book, page, locationState, currentPath} = getProps();
-  const href = anchor.getAttribute('href');
+export const contentLinkHandler = (anchor: HTMLAnchorElement, getProps: () => ContentLinkProp) =>
+  async(e: MouseEvent) => {
+    const {
+      references,
+      navigate,
+      book,
+      page,
+      locationState,
+      currentPath,
+      hasUnsavedHighlight,
+      focusedHighlight,
+    } = getProps();
+    const href = anchor.getAttribute('href');
 
-  if (!href || !book || !page || e.metaKey) {
-    return;
-  }
+    if (!href || !book || !page || isClickWithModifierKeys(e)) {
+      return;
+    }
 
-  const {hash, search, pathname} = new URL(href, assertWindow().location.href);
-  const reference = references.find(isPathRefernceForBook(pathname, book));
+    const {hash, search, pathname} = new URL(href, assertWindow().location.href);
+    const reference = references.find(isPathRefernceForBook(pathname, book));
+
+    if (!reference && !(pathname === currentPath && hash)) {
+      return;
+    }
 
-  if (reference) {
-    e.preventDefault();
-    // defer to allow other handlers to execute before nav happens
-    defer(() => navigate({
-      params: reference.params,
-      route: content,
-      state: {
-        ...locationState,
-        ...reference.state,
-      },
-    }, {hash, search}));
-  } else if (pathname === currentPath && hash) {
     e.preventDefault();
-    // defer to allow other handlers to execute before nav happens
-    defer(() => navigate({
-      params: getBookPageUrlAndParams(book, page).params,
-      route: content,
-      state: {
-        ...locationState,
-        ...getBookPageUrlAndParams(book, page).state,
 
-      },
-    }, {hash, search}));
-  }
-};
+    const containedHighlight = getContainedHighlightNode(anchor);
```

Comment:
its possible that the user highlights part of a link, and they then click the link but not the highlight. what we want to check for is if the user actually clicked the highlight and in that case not process the link.

i had thought doing the stopPropagation from the highlight click handler would be good, but i see now that the highlighter event listener is on the container, while the link listeners are on the links, so the links get evaluated first.

first of all, this logic doesn't do well in the case that a highlight does not span the entire link, its possible to click on a link that has a highlight in it, but not actually click on the highlight, in the code that is written here, those clicks would just do nothing. similarly, its possible to have _two_ separate highlights in a link, and with this code you would never be able to click the link, because one of them would always be unfocused.

i'm not loving the way the link handler is aware of the highlights logic, but i don't really have a better idea right now. maybe the best we can do for now is to rewrite your `getContainedHighlightNode` to accept `event.target` and see if that is inside a highlight, and if its not focused, not handle the link